### ISSUE TITLE
How does generator update?

### ISSUE THREAD

**Issue by Jessielym66** [DeepGraphLearning/RNNLogic#10]
I'm confused about the EM step in codes. The paper describes that the rule generator is updated by maximizing the log-likelyhood of the identified high-quality rules,  while in the codes, rule generator computes loss between 'logits' and 'target'. I wonder that whether the 'logits' is  the first term 'RNN(rule|r)' in Eq.(9)? And what does the 'target' mean? Why do you apply CrossEntropyLoss on them? Does it achieve the efficiency same as maximization? Or do I misunderstand the code?
